inference-engine/src/hetero_plugin/hetero_plugin.cpp

```diff
@@ -50,8 +56,40 @@ InferenceEngine::ExecutableNetworkInternal::Ptr Engine::LoadExeNetworkImpl(const
     if (GetCore() == nullptr) {
         THROW_IE_EXCEPTION << "Please, work with HETERO device via InferencEngine::Core object";
     }
-
-    return std::make_shared<HeteroExecutableNetwork>(*cloneNet(network), mergeConfigs(_config, config), this);
+    auto tconfig = mergeConfigs(_config, config);
+    auto it = tconfig.find("TARGET_FALLBACK");
+    if (it == tconfig.end()) {
+        THROW_IE_EXCEPTION << "The 'TARGET_FALLBACK' option was not defined for heterogeneous plugin";
+    }
+    DeviceMetaInformationMap metaDevices = GetDevicePlugins(it->second, tconfig);
+
+    auto function = network.getFunction();
+    if (function != nullptr) {
+        auto anyDeviceDoNotSupportNgraph =
+        std::any_of(std::begin(metaDevices), std::end(metaDevices),
+                    [&] (const DeviceMetaInformationMap::value_type& metaDevice) {
+                        auto& deviceName = metaDevice.first;
+                        auto clonedNetwork = cloneNetwork(network);
+                        GetCore()->QueryNetwork(*clonedNetwork, deviceName, metaDevice.second);
+                        return (clonedNetwork->getFunction() == nullptr);
+                    });
+        if (anyDeviceDoNotSupportNgraph) {
+            auto clonedNetwork = cloneNetwork(network);
+            auto function = clonedNetwork->getFunction();
+            ::ngraph::op::GenericIE::DisableReshape noReshape(function);
+            ::ngraph::pass::CommonOptimizations().run_on_function(function);
+            ::ngraph::pass::ConvertOpSet3ToOpSet2().run_on_function(function);
+            ::ngraph::pass::ConvertOpSet2ToOpSet1().run_on_function(function);
+            ::ngraph::pass::ConvertOpSet1ToLegacy().run_on_function(function);
+            return std::make_shared<HeteroExecutableNetwork>(
+                *InferenceEngine::details::convertFunctionToICNNNetwork(function, *clonedNetwork),
```

Comment:
@ilyachur Is it correct way to convert to legacy representation?

Reply:
Yes it is correct. First of all you need transform all operations to legacy and after that convert to old representation. It is correct!